### PR TITLE
fix flicker on "Select" elements

### PR DESCRIPTION
**Pull request by danielzsh**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [x] I have linked this PR to any issues that it closes.

### CODE REVIEW

**Comment by danielzsh** [comment]
This solution seems to work, but idk if its valid 😅

**Comment by danielzsh** [comment]
wait nvm, i forgot to test `yarn build` 😭

File: src/context/DarkModeContext.tsx [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
 
 // Note: this is separate from DarkModeProvider.tsx because this is imported by Storybook.
 
 export const DarkModeContext = React.createContext<boolean>(null);
 
 export const useDarkMode = (): boolean => {
   const context = React.useContext(DarkModeContext);
+  if (!useIsUserDataLoaded()) {
+    return document.documentElement.classList.contains('dark');
+  }
```
Comment (thecodingwizard) [inline src/context/DarkModeContext.tsx:12]: Maybe this should be moved into `DarkModeProvider`?

Note that any access to document must be inside a `useEffect`. I think a better approach would be to leave this function alone and modify `DarkModeProvider` instead. DarkModeProvider waits for the user data to load before setting dark mode, which is slow. Instead, before `isLoaded` is set, it should do something similar to `gatsby-ssr.tsx` (ie. try loading from `themeKey` or default to system).

**Review by thecodingwizard — APPROVED** [review]
lgtm other than the comment I left!

File: src/context/DarkModeContext.tsx (hunk 1 above)
Comment (danielzsh) [reply src/context/DarkModeContext.tsx]: Curious; even though the first value is likely still wrong, there's no flicker at all anymore? Do you know what's going on?

Comment (danielzsh) [reply src/context/DarkModeContext.tsx]: Ah I see, it's probably because everything on the problems page is only rendered client side 😅

**Comment by danielzsh** [comment]
Btw, would you mind re-approving? I think any commits made after an approving review auto-dismisses it as stale, which is kinda annoying...

**Review by thecodingwizard — APPROVED** [review]

**Pull request merged**